RELEASE_CHECKLIST.md

```diff
@@ -4,16 +4,18 @@
 
 This document describes the checklist to publish a release via GitHub workflow.
 
-NOTE: Make sure the dependencies in `go.mod` file are expected by the release. For example, if there are dependencies on certain version of notation library (notation-go or notation-core-go) or ORAS library (oras-go), make sure that version of library is released first, and the version number is updated accordingly in `go.mod` file.  After updating go.mod file, run `go mod tidy` to ensure the go.sum file is also updated with any potential changes.
-
 ## Release Process
 
+1. Check if there are any security vulnerabilities fixed and security advisories published before a release. Security advisories should be linked on the release notes
 1. Determine a [SemVer2](https://semver.org/)-valid version prefixed with the letter `v` for release. For example, `version="v1.0.0-alpha.1"`.
-1. Bump up the `Version` in [internal/version/version.go](internal/version/version.go#L5) and open a PR for the changes.
-1. Wait for the PR merge.
-1. Be on the main branch connected to the actual repository (not a fork) and `git pull`.  Ensure `git log -1` shows the latest commit on the main branch.
-1. Create a tag `git tag -am $version $version`
-1. `git tag` and ensure the name in the list added looks correct, then push the tag directly to the repository by `git push --follow-tags`.
+1. Create an issue to vote for the new release. Add the link of change logs and determine a specific git commit to tag on the issue description. List all repo-level maintainers and make sure have majority of approvals from the maintainers before releasing it. 
+1. If there are code changes in notation-go or notation-core-go library, follow the three steps above and cut the release for the library. Then submit a PR to update the dependency versions in notation [go.mod](go.mod) and [go.sum](go.sum). Run `go mod tidy` to ensure the `go.sum` file is also updated with any potential changes.

```

Comment:
There are 3 different `go.mod` file in notation pkg. Can we please list all the files for each package where changes needs to made and better if we can write commands?

Reply:
@priteshbandi  This checklist is currently used for the Notation CLI repo only. There is another plan to update the release checklist and process document by standing on the whole Notation project's point of view and involve notation-go and notation-core-go projects in a central place. see https://github.com/notaryproject/.github/issues/28. That said we will create a doc in `.github` repo and consolidate the release process for notation, notation-go, notation-core-go, notaryproject(spec).

Is it acceptable?

Reply:
@FeynmanZhou it seems still in `notation`  repo, for example, `go.mod` for e2e testing needs to be updated as well.

Reply:
@yizha1  Good catch. Add https://github.com/notaryproject/notation/blob/main/test/e2e/go.mod to this step